WorkloadManagementSystem/Client/WMSClient.py

```diff
@@ -130,21 +135,56 @@ def submitJob( self, jdl, jobDescriptionObject = None ):
       jdlString = "[%s]" % jdlString
     classAdJob = ClassAd( jdlString )
     if not classAdJob.isOK():
-      return S_ERROR( 'Invalid job JDL' )
+      return S_ERROR( EWMSJDL, 'Invalid job JDL' )
 
     # Check the size and the contents of the input sandbox
     result = self.__uploadInputSandbox( classAdJob, jobDescriptionObject )
     if not result['OK']:
       return result
 
     # Submit the job now and get the new job ID
+    bulkTransaction = self.operationsHelper.getValue( 'JobScheduling/BulkSubmissionTransaction', False )
+    if bulkTransaction:
+      result = getParameterVectorLength( classAdJob )
+      if not result['OK']:
+        return result
+      nJobs = result['Value']
+      parametricJob = nJobs > 0
+
     if not self.jobManager:
       self.jobManager = RPCClient( 'WorkloadManagement/JobManager',
                                    useCertificates = self.useCertificates,
                                    timeout = self.timeout )
+
     result = self.jobManager.submitJob( classAdJob.asJDL() )
-    if 'requireProxyUpload' in result and result['requireProxyUpload']:
+
+    if bulkTransaction and parametricJob:
+      gLogger.debug( 'Applying transactional job submission' )
+      # The server indeed applies transactional bulk submission, we should confirm the jobs
+      if result['OK'] and result.get( 'requireBulkSubmissionConfirmation' ):
+        jobIDList = result['Value']
+        if len( jobIDList ) == nJobs:
+          # Confirm the submitted jobs
+          confirmed = False
+          for attempt in xrange(3):
```

Comment:
`for _ in xrange(3)`

Reply:
I guess this is for pylint. Can I do for _attempt_ in xrange(3) ?

Reply:
I just want that the code reads "make 3 attempts to do that" without any thinking by the reader. _ is a throwaway variable for somebody python savvy